ocaml/typing/typecore.ml

```diff
@@ -8598,38 +8582,14 @@ and type_mode_expr
   : Jane_syntax.Modes.expression -> _ = function
   | Coerce (m, sbody) ->
     let modes = Typemode.transl_mode_annots m in
-    (* CR zqian: All axes should be handled in one go. We can't do that yet
-       because [mode_strictly_local] is special. *)
-    let expected_mode =
-      match modes.uniqueness with
-      | Some Unique ->
-          let expected_mode = mode_unique expected_mode in
-          expect_mode_cross env ty_expected expected_mode
-      | Some m ->
-          Misc.fatal_errorf "The mode %a should not interpret" Uniqueness.Const.print m
-      | None -> expected_mode
-    in
-    let expected_mode =
-      match modes.linearity with
-      | Some Once ->
-          let expected_mode = expect_mode_cross env ty_expected expected_mode in
-          submode ~loc ~env ~reason:Other
-            (Value.min_with_linearity Linearity.once) expected_mode;
-          mode_once expected_mode
-      | Some m ->
-          Misc.fatal_errorf "The mode %a should not interpret" Linearity.Const.print m
-      | None -> expected_mode
-    in
+    let min = Alloc.Const.Option.value ~default:Alloc.Const.min modes |> Const.alloc_as_value in
```

Comment:
Hmm what do you mean by `Value.Const`?

Reply:
Well, if we have `Alloc.Const`, we could have `Value.Const`. Maybe it doesn't exist today, but it should. (And will, after https://github.com/riaqn/flambda-backend/pull/3 anyway.)

Reply:
I think `Alloc.Const` is used to interact with user-written mode annotations, and `Value.Const` would make less sense, since users can't write `regional`.

Reply:
Yes that's true, but it would at least be useful (and faster!) here. If you don't want to do this here, please leave a `CR modes` comment and then I'll try to remember to do this after riaqn#3.

Reply:
OK after a closer look I don't think we should do this. `transl_mode_annots` returns some `Alloc.Const.Option.t`, and we need some `Value.Const.t`. Currently we convert that to `Alloc.Const.t` and then `Value.Const.t`.

The optimisation you are proposing is to jump from `Alloc.Const.Option.t` to `Value.Const.t`. It's extra code which is the composition of two existing canonical functions; it won't be automatically generated by riaqn#3, and the performance gain is marginal.